cpp/ql/src/Security/CWE/CWE-327/BrokenCryptoAlgorithm.ql

```diff
@@ -48,7 +48,9 @@ class InsecureMacroSpec extends InsecureCryptoSpec, MacroInvocation {
     exists(this.getAGeneratedElement().(ControlFlowNode)) and
     // exclude expressions controlling ifs/switches (as they may not be used).
     not any(IfStmt c).getCondition().getAChild*() = this.getAGeneratedElement() and
-    not any(SwitchCase c).getExpr().getAChild*() = this.getAGeneratedElement()
+    not any(SwitchCase c).getExpr().getAChild*() = this.getAGeneratedElement() and
+    // exclude expressions in array initializers (as they may not be used).
+    not any(AggregateLiteral i).getAChild*() = this.getAGeneratedElement() 
```

Comment:
Instead of having an exclusion list for which constructs we consider, would it be better to have an inclusion list instead? Something like "only `MacroInvocations` that are arguments to `FunctionCall`s"?

Reply:
Yeah, I see that you ended up rewriting all of these 👍 The final version LGTM.